sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs

```diff
@@ -154,38 +194,114 @@ private static string ReadAndVerifyEnvironmentVariable(string variableName)
+            return CreateNamespace();
+        }
+
+        /// <summary>
+        ///   It instantiates a new namespace starting from the connection string passed in.
+        /// </summary>
+        ///
+        /// <returns>The active Event Hubs namespace for this test run.</returns>
+        ///
+        private static EventHubScope.NamespaceProperties PopulateNamespacePropertiesFromConnectionString(string connectionString) =>
+            EventHubScope.PopulateNamespacePropertiesFromConnectionString(connectionString);
+
+        /// <summary>
+        ///   Parses a well-formed connection string to extract the fully qualified namespace.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to parse.</param>
+        ///
+        /// <returns>The fully qualified namespace contained in the connection string if the connection string is well-formed; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        private static string ParseFullyQualifiedNamespace(string connectionString) =>
+            new Uri(ConnectionStringTokenParser.ParseTokenAndReturnValue(connectionString, "Endpoint")).Host;
+
+        /// <summary>
+        ///   Parses a well-formed connection string to extract the shared access key name.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to parse.</param>
+        ///
+        /// <returns>The fully qualified namespace contained in the connection string if the connection string is well-formed; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        private static string ParseSharedAccessKeyName(string connectionString) =>
+            ConnectionStringTokenParser.ParseTokenAndReturnValue(connectionString, "SharedAccessKeyName");
 
-            if (end < 0)
+        /// <summary>
+        ///   Parses a well-formed connection string to extract the shared access key.
+        /// </summary>
+        /// <param name="connectionString"></param>
+        ///
+        /// <returns>The fully qualified namespace contained in the connection string if the connection string is well-formed; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        private static string ParseSharedAccessKey(string connectionString) =>
+            ConnectionStringTokenParser.ParseTokenAndReturnValue(connectionString, "SharedAccessKey");
+
+        /// <summary>
+        ///   Reads a connection string from environment.
+        ///   If any is found it removes the entity path from it.
+        /// </summary>
+        ///
+        /// <param name="environmentVariable">The name of the environment variable containing the connection string.</param>
+        ///
+        /// <returns>The connection string without the entity path.</returns>
+        ///
+        private static string ParseConnectionString(string environmentVariable)
```

Comment:
The name for this is potentially confusing since it's not just parsing, it's also reading the environment variable.   I looked at the call to this and thought there was a typo because it was an environment variable name.   

Maybe consider revising to `ReadConnectionStringFromEnvironment` or similar?